languages/rust/oso/src/host/method.rs

```diff
@@ -83,3 +267,346 @@ where
+where
+    F: Fn(&Receiver, A, B, C, D, E, G, H, I, J, K, L, M) -> R + Send + Sync,
+{
+    type Result = R;
+
+    fn invoke(
+        &self,
+        receiver: &Receiver,
+        args: (A, B, C, D, E, G, H, I, J, K, L, M),
+    ) -> Self::Result {
+        (self)(
+            receiver, args.0, args.1, args.2, args.3, args.4, args.5, args.6, args.7, args.8,
+            args.9, args.10, args.11,
+        )
+    }
+}
+
+impl<A, B, C, D, E, G, H, I, J, K, L, M, N, F, R, Receiver>
+    Method<Receiver, (A, B, C, D, E, G, H, I, J, K, L, M, N)> for F
+where
+    F: Fn(&Receiver, A, B, C, D, E, G, H, I, J, K, L, M, N) -> R + Send + Sync,
+{
+    type Result = R;
+
+    fn invoke(
+        &self,
+        receiver: &Receiver,
+        args: (A, B, C, D, E, G, H, I, J, K, L, M, N),
+    ) -> Self::Result {
+        (self)(
+            receiver, args.0, args.1, args.2, args.3, args.4, args.5, args.6, args.7, args.8,
+            args.9, args.10, args.11, args.12,
+        )
+    }
+}
+
+impl<A, B, C, D, E, G, H, I, J, K, L, M, N, O, F, R, Receiver>
+    Method<Receiver, (A, B, C, D, E, G, H, I, J, K, L, M, N, O)> for F
+where
+    F: Fn(&Receiver, A, B, C, D, E, G, H, I, J, K, L, M, N, O) -> R + Send + Sync,
+{
+    type Result = R;
+
+    fn invoke(
+        &self,
+        receiver: &Receiver,
+        args: (A, B, C, D, E, G, H, I, J, K, L, M, N, O),
+    ) -> Self::Result {
+        (self)(
+            receiver, args.0, args.1, args.2, args.3, args.4, args.5, args.6, args.7, args.8,
+            args.9, args.10, args.11, args.12, args.13,
+        )
+    }
+}
+
+impl<A, B, C, D, E, G, H, I, J, K, L, M, N, O, P, F, R, Receiver>
+    Method<Receiver, (A, B, C, D, E, G, H, I, J, K, L, M, N, O, P)> for F
+where
+    F: Fn(&Receiver, A, B, C, D, E, G, H, I, J, K, L, M, N, O, P) -> R + Send + Sync,
+{
+    type Result = R;
+
+    fn invoke(
+        &self,
+        receiver: &Receiver,
+        args: (A, B, C, D, E, G, H, I, J, K, L, M, N, O, P),
+    ) -> Self::Result {
+        (self)(
+            receiver, args.0, args.1, args.2, args.3, args.4, args.5, args.6, args.7, args.8,
+            args.9, args.10, args.11, args.12, args.13, args.14,
+        )
+    }
+}
+
+impl<A, B, C, D, E, G, H, I, J, K, L, M, N, O, P, Q, F, R, Receiver>
+    Method<Receiver, (A, B, C, D, E, G, H, I, J, K, L, M, N, O, P, Q)> for F
+where
+    F: Fn(&Receiver, A, B, C, D, E, G, H, I, J, K, L, M, N, O, P, Q) -> R + Send + Sync,
+{
+    type Result = R;
+
+    fn invoke(
+        &self,
+        receiver: &Receiver,
+        args: (A, B, C, D, E, G, H, I, J, K, L, M, N, O, P, Q),
+    ) -> Self::Result {
+        (self)(
+            receiver, args.0, args.1, args.2, args.3, args.4, args.5, args.6, args.7, args.8,
+            args.9, args.10, args.11, args.12, args.13, args.14, args.15,
+        )
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    // This would be great if it was a proc macro but I don't wanna set up another crate.
```

Comment:
This seems like a decent solution:
https://docs.rs/impl-trait-for-tuples/0.1.3/impl_trait_for_tuples/

Reply:
Might not work for method though since the trait isn't on the tuple itself, it's the assoc type.